### PR TITLE
feat: move mev.rs from reth to rpc-types-mev

### PR DESCRIPTION
**Pull request by 0xqd**
Ref https://github.com/paradigmxyz/reth/issues/9027

### CODE REVIEW

File: crates/rpc-types-mev/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+exclude.workspace = true
+
+[package.metadata.docs.rs]
+all-features = true
+rustdoc-args = ["--cfg", "docsrs"]
+
+[dependencies]
+alloy-eips = { workspace = true, features = ["serde"] }
+alloy-primitives.workspace = true
+alloy-rpc-types.workspace = true
```
Comment (mattsse) [inline crates/rpc-types-mev/Cargo.toml:21]: this should directly depend on alloy-rpc-types-eth

**Review by mattsse — CHANGES_REQUESTED** [review]
ty, some early comments

we also want a feature in rpc-types for this

File: crates/rpc-types/Cargo.toml [hunk 2]
```diff
@@ -25,6 +25,7 @@ alloy-rpc-types-anvil = { workspace = true, optional = true }
 alloy-rpc-types-beacon = { workspace = true, optional = true }
 alloy-rpc-types-engine = { workspace = true, optional = true }
 alloy-rpc-types-eth = { workspace = true, optional = true }
+alloy-rpc-types-mev = { workspace = true, optional = true }
```
Comment (mattsse) [inline crates/rpc-types/Cargo.toml:28]: needs a feature and reexport in lib.rs

**Review by mattsse — CHANGES_REQUESTED** [review]
I'd like to not have a ton of modules just for single types

I'm fine with separating eth and mev modules

**Comment by 0xqd** [comment]
yeah, it's reasonable, I tried to split it by endpoint but it seems to be a bit much at the end.

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**